packages/xsnap/src/avaXS.js

```diff
@@ -0,0 +1,209 @@
+  await withXSnap({ name: input, handleCommand }, async worker => {
+    await worker.evaluate(pathGlobalsKludge);
+    await worker.issueStringCommand(src);
+  });
+
+  return { qty, byStatus };
+}
+
+/** @type {(argv: string[], name: string) => boolean } */
+function extractFlag(argv, name) {
+  const ix = argv.indexOf(name);
+  if (ix >= 0) {
+    argv.splice(ix, 1);
+    return true;
+  }
+  return false;
+}
+
+/**
+ * @param {string[]} argv
+ * @param {{
+ *   bundleSource: typeof import('@agoric/bundle-source').default,
+ *   spawn: typeof import('child_process')['spawn'],
+ *   osType: typeof import('os')['type'],
+ *   readFile: typeof import('fs')['promises']['readFile'],
+ *   resolve: typeof import('path').resolve,
+ *   dirname: typeof import('path').dirname,
+ * }} io
+ */
+async function main(
+  argv,
+  { bundleSource, spawn, osType, readFile, resolve, dirname },
+) {
+  async function testSource(input) {
+    const bundle = await bundleSource(input, 'getExport', {
+      externals: [
+        'ava',
+        '@agoric/bundle-source',
+        '@agoric/install-ses',
+        '@agoric/install-metering-and-ses',
+      ],
+    });
+    return bundle.source;
+  }
+
+  const sesShim = await asset(`../dist/bootstrap.umd.js`, readFile);
+  const tinyAva = await asset(`./avaAssertXS.js`, readFile);
+  const avaHandler = await asset(`./avaHandler.js`, readFile);
+
+  const debug = extractFlag(argv, '--debug');
+
+  // we only use import() in type annotations
+  const hideImport = (/** @type { string } */ src) =>
+    src.replace(/import\(/g, '');
+
+  async function withXSnap(opts, thunk) {
+    const worker = xsnap({
+      meteringLimit: 0,
+      spawn,
+      os: osType(),
+      stdout: 'inherit',
+      stderr: 'inherit',
+      debug,
+      ...opts,
+    });
+    try {
+      await worker.evaluate(sesShim);
+      await worker.evaluate(hideImport(tinyAva));
+      await worker.evaluate(hideImport(avaHandler));
+      return await thunk(worker);
+    } finally {
+      worker.terminate();
+    }
+  }
+
+  let totalTests = 0;
+  const stats = { ok: 0, 'not ok': 0, SKIP: 0 };
+  for (const input of argv.slice(2)) {
+    console.log('# test script:', input);
+    const src = await testSource(input);
+
+    const { qty, byStatus } = await runTest(input, src, {
+      withXSnap,
+      bundleSource,
+      resolve,
+      dirname,
+      debug,
+    });
+
+    totalTests += qty;
+    Object.entries(byStatus).forEach(([status, q]) => {
+      stats[status] += q;
+    });
+  }
+
+  console.log({ totalTests, stats });
+  return stats['not ok'] > 0 ? 1 : 0;
+}
+
+/* eslint-disable global-require */
+if (require.main === module) {
+  main(process.argv, {
+    bundleSource: require('@agoric/bundle-source').default,
+    spawn: require('child_process').spawn,
+    osType: require('os').type,
+    readFile: require('fs').promises.readFile,
+    resolve: require('path').resolve,
+    dirname: require('path').dirname,
+  })
+    .then(status => {
```

Comment:
Of all the weird layouts prettier foists on us, this newline between the close paren and the dot is the most surprising. But I tried and indeed prettier does restore it.